### PR TITLE
test(asyncio): replace asyncio.coroutine with async def

### PR DESCRIPTION
**Pull request by jd**
This introduces usage of pytest-asyncio to run the async test functions.

### CODE REVIEW

File: tests/contrib/asyncio/test_tracer_safety.py [hunk 1]
```diff
@@ -50,19 +47,21 @@ def test_trace_coroutine(self):
...
+            await asyncio.sleep(0.01)
+
+    futures = [asyncio.ensure_future(coro()) for x in range(1000)]
+    for future in futures:
+        await future
+
+    # the trace is wrong but the Context is finished
+    traces = tracer.pop_traces()
+    assert 1000 == len(traces)
+    assert 1 == len(traces[0])
```
Comment (jd) [inline tests/contrib/asyncio/test_tracer_safety.py:67]: I had to change the result here.
I'm not sure if the comment is correct or if my change is wrong?
@Kyle-Verhoog can you weigh in?

Comment (jd) [reply tests/contrib/asyncio/test_tracer_safety.py:67]: It's actually weirder: there is now 1k traces with Python 3.9, but Python 3.5 still has the previous 1 traces.

**Comment by jd** [comment]
@Kyle-Verhoog there are still some test failure on Python 3.5 and I'm wondering if it's not a bug where the code wouldn't be using the right loop.
If you have an opinion on this :)

**Comment by Kyle-Verhoog** [comment]
Ahh yeah the asyncio integration only uses the active event loop at the time `patch()` is called. It looks like pytest-asyncio is using a new event loop for each test case.

Surprisingly I don't think we've heard any feedback from users about this issue. This also becomes a non-issue in Python 3.7+ with #1936 since we don't need to patch the event loop.

@jd I don't think we have to do anything immediate here. To work around it we just need to call `ddtrace.contrib.asyncio.patch()` before each test case to work around it for now.

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**